File: frontend/src/types/formulario.ts
export interface Titular {
  nombre: string;
  nit: string;
  representante_legal: string;
  direccion: string;
}

export interface Predio {
  nombre: string;
  municipio: string;
  vereda: string;
  latitud: string;
  longitud: string;
}

export interface Aprovechamiento {
  tipo: string;
  justificacion: string;
  volumen_total: number;
  unidad: string;
}

export interface Especie {
  nombre: string;
  cantidad: number;
  diametro_cm: number | null;
}

export interface FormData {
  titular: Titular;
  predio: Predio;
  aprovechamiento: Aprovechamiento;
  especies: Especie[];
}

export interface NormalizedData {
  municipio: {
    nombre_oficial: string;
    confianza: number;
    codigo_dane: string;
    requiere_confirmacion: boolean;
  };
  latitud: { valor_decimal: string; eje: string; error?: boolean; en_colombia?: boolean };
  longitud: { valor_decimal: string; eje: string; error?: boolean; en_colombia?: boolean };
  tipo_aprovechamiento: { categoria: string; error?: boolean };
  autoridad: { nombre: string; sigla: string; error?: boolean };
  especies: Array<Especie & { normalizacion: {
    nombre_comun: string;
    nombre_cientifico: string;
    metodo: string;
    confianza?: number;
    requiere_revision?: boolean;
    error?: string;
  } }>;
  requisitos?: { autoridad: string; anexos: string[] };
  requiere_revision?: boolean;
  revision_confirmada?: boolean;
  listo_para_generar: boolean;
}

File: backend/app/schemas/formulario.py
from pydantic import BaseModel, Field
from typing import Any, Optional, List

class Titular(BaseModel):
    nombre: str
    nit: str
    representante_legal: str
    direccion: str

class Predio(BaseModel):
    nombre: str
    municipio: str
    vereda: Optional[str] = None
    latitud: str
    longitud: str

class Aprovechamiento(BaseModel):
    tipo: str
    justificacion: str
    volumen_total: float
    unidad: str

class Especie(BaseModel):
    nombre: str
    cantidad: int
    diametro_cm: Optional[float] = None

class FormularioCompleto(BaseModel):
    titular: Titular
    predio: Predio
    aprovechamiento: Aprovechamiento
    especies: List[Especie]
    autoridad_seleccionada: Optional[str] = None
    detalles_autoridad: dict[str, Any] = Field(default_factory=dict)
    confirmar_revision: bool = False